packages/expo-modules-core/common/cpp/EventEmitter.cpp

```diff
@@ -0,0 +1,139 @@
+    object.setNativeState(runtime, state);
+    return state;
+  }
+  return nullptr;
+}
+
+#pragma mark - Utils
+
+jsi::Function getClass(jsi::Runtime &runtime) {
+  return runtime
+    .global()
+    .getPropertyAsObject(runtime, "expo")
+    .getPropertyAsFunction(runtime, "EventEmitter");
+}
+
+void installClass(jsi::Runtime &runtime) {
+  jsi::Function eventEmitterClass = common::createClass(runtime, "EventEmitter");
+  jsi::Object prototype = eventEmitterClass.getPropertyAsObject(runtime, "prototype");
+
+  jsi::HostFunctionType addListener = [](jsi::Runtime &runtime, const jsi::Value &thisValue, const jsi::Value *args, size_t count) -> jsi::Value {
+    std::string eventName = args[0].asString(runtime).utf8(runtime);
+    jsi::Function listener = args[1].asObject(runtime).asFunction(runtime);
+    jsi::Object thisObject = thisValue.getObject(runtime);
+
+    if (NativeState::Shared state = NativeState::get(runtime, thisObject, true)) {
+      state->listeners.add(runtime, eventName, listener);
+    }
+    return jsi::Value::undefined();
+  };
+
+  jsi::HostFunctionType removeListener = [](jsi::Runtime &runtime, const jsi::Value &thisValue, const jsi::Value *args, size_t count) -> jsi::Value {
+    std::string eventName = args[0].asString(runtime).utf8(runtime);
+    jsi::Function listener = args[1].asObject(runtime).asFunction(runtime);
+    jsi::Object thisObject = thisValue.getObject(runtime);
+
+    if (NativeState::Shared state = NativeState::get(runtime, thisObject, false)) {
+      state->listeners.remove(runtime, eventName, listener);
+    }
+    return jsi::Value::undefined();
+  };
+
+  jsi::HostFunctionType removeListeners = [](jsi::Runtime &runtime, const jsi::Value &thisValue, const jsi::Value *args, size_t count) -> jsi::Value {
+    std::string eventName = args[0].asString(runtime).utf8(runtime);
+    jsi::Object thisObject = thisValue.getObject(runtime);
+
+    if (NativeState::Shared state = NativeState::get(runtime, thisObject, false)) {
+      state->listeners.removeAll(eventName);
+    }
+    return jsi::Value::undefined();
+  };
+
+  jsi::HostFunctionType emit = [](jsi::Runtime &runtime, const jsi::Value &thisValue, const jsi::Value *args, size_t count) -> jsi::Value {
+    std::string eventName = args[0].asString(runtime).utf8(runtime);
+    jsi::Object thisObject = thisValue.getObject(runtime);
+
+    if (NativeState::Shared state = NativeState::get(runtime, thisObject, false)) {
+      // Pass the arguments without the first that is the event name.
+      state->listeners.call(runtime, eventName, thisObject, &args[1], count - 1);
+    }
+    return jsi::Value::undefined();
+  };
+
+  jsi::PropNameID addListenerProp = jsi::PropNameID::forAscii(runtime, "addListener", 11);
+  jsi::PropNameID removeListenerProp = jsi::PropNameID::forAscii(runtime, "removeListener", 14);
+  jsi::PropNameID removeListenersProp = jsi::PropNameID::forAscii(runtime, "removeListeners", 15);
```

Comment:
Maybe `removeAllListeners` for clarity and being compatible with the current JS emitter?
```suggestion
  jsi::PropNameID removeListenersProp = jsi::PropNameID::forAscii(runtime, "removeAllListeners", 15);
```